src/Aspire.Hosting.Azure.ContainerApps/AzureContainerAppContainerExtensions.cs

```diff
@@ -0,0 +1,48 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+using Aspire.Hosting.ApplicationModel;
+using Azure.Provisioning.AppContainers;
+
+namespace Aspire.Hosting.Azure;
+
+/// <summary>
+/// Provides extension methods for publishing container resources as container apps in Azure.
+/// </summary>
+public static class AzureContainerAppContainerExtensions
+{
+    /// <summary>
+    /// Publishes the specified container resource as a container app.
+    /// </summary>
+    /// <typeparam name="T">The type of the container resource.</typeparam>
+    /// <param name="container">The container resource builder.</param>
+    /// <param name="configure">The configuration action for the container app.</param>
+    /// <returns>The updated container resource builder.</returns>
+    /// <remarks>
+    /// This method checks if the application is in publish mode. If it is, it adds the necessary infrastructure
+    /// for container apps and applies the provided configuration action to the container app.
+    /// </remarks>
+    /// <example>
+    /// <code>
+    /// var builder = new ResourceBuilder&lt;MyContainerResource&gt;();
+    /// builder.PublishAsContainerApp((module, app) =>
+    /// {
+    ///     // Configure the container app here
+    /// });
+    /// </code>
+    /// </example>
+#pragma warning disable AZPROVISION001 // Type is for evaluation purposes only and is subject to change or removal in future updates. Suppress this diagnostic to proceed.
+    public static IResourceBuilder<T> PublishAsContainerApp<T>(this IResourceBuilder<T> container, Action<ResourceModuleConstruct, ContainerApp> configure) where T : ContainerResource
```

Comment:
Should the configure action be `nullable`?

Reply:
> Also, I thought you wanted to move away from these callbacks, and instead:

We can't because this isn't an AzureConstructResource. Also there's no app model resource that you need to reference here. This is a nested "deployment target" resource that you get access to.

> Should the configure action be nullable?

We could. That could be the way you opt in as well but it feels strange. I like the pattern of, call the method globally to opt in or customize a resource to opt in.

Reply:
Is there value in marking this experimental until the CDK isn't'?

Reply:
> Is there value in marking this experimental until the CDK isn't'?

Why do we have all our other APIs marked as experimental then? Should they also not have the attribute?

Reply:
Do we?

Reply:
Done